roles/dd-nodepool/files/nodepool_server.py

```diff
@@ -55,3 +70,25 @@ def get_nodepool_status(self, status_url, timeout, service_check_tags):
                            AgentCheck.OK,
                            message='Received status from %s' % status_url,
                            tags=service_check_tags)
+
+    def get_nodepool_images(self, images_url, timeout, service_check_tags):
+        r = requests.get(images_url, timeout=timeout)
+        currentTime = int(time.time())
+        staleImage = False
+
+        for image in r:
```

Comment:
You're iterating through the lines of the body here.  If you update the endpoint and request the json, you can instead do:

```
for image in r.json():
```

... which'll iterate instead through image dicts below.